app/Http/Services/Tabbar/TabbarService.php

```diff
@@ -0,0 +1,138 @@
+            'CONTENTS' => $this->_getContentsTabInfo($tabbar['contents'], $pathInfo->getContentsTabUrlInfo()),
+            'DOCS' => $this->_getDocsTabInfo($tabbar['docs'], $pathInfo->getDocsTabUrlInfo()),
+            'DEV' => $this->_getDevTabInfo($tabbar['dev'], $pathInfo->getDevTabUrlInfo())
+        ];
+    }
+
+    /**
+     * Homeタブの情報を取得する
+     * 
+     * @param string $homeParam リクエストパラメータのhomeの値
+     * @param array $homeTabUrlInfo Homeタブ内に表示するページのURL一覧
+     * @return string Homeタブに表示するページのURL
+     * 
+     */
+    private function _getHomeTabInfo($homeParam, $homeTabUrlInfo) {
+
+        // パラメータからHomeタブのlistのキー値を作成する
+        $listkey = strtoupper($homeParam);
+
+        // homeパラメータ値がhomeタブに存在するurlと一致しない場合
+        if (!array_key_exists($listkey, $homeTabUrlInfo)) {
+            // home_topを表示させる
+            return $homeTabUrlInfo['TOP'];
+        } else {
+            return $homeTabUrlInfo[$listkey];
+        }
+    }
+
+    /**
+     * Contentsタブの情報を取得する
+     * 
+     * @param string $contentsParam リクエストパラメータのcontentsの値
+     * @param array $contentsTabUrlInfo Contentsタブ内に表示するページのURL一覧
+     * @return string Contentsタブに表示するページのURL
+     * 
+     */
+    private function _getContentsTabInfo($contentsParam, $contentsTabUrlInfo) {
+
+        // パラメータからContentsタブのlistのキー値を作成する
+        $listkey = strtoupper($contentsParam);
+
+        // homeパラメータ値がContentsタブに存在するurlと一致しない場合
+        if (!array_key_exists($listkey, $contentsTabUrlInfo)) {
+            // contents_topを表示させる
+            return $contentsTabUrlInfo['TOP'];
+        } else {
+            return $contentsTabUrlInfo[$listkey];
+        }
+    }
+
+    /**
+     * Devタブの情報を取得する
+     * 
+     * @param string $devParam リクエストパラメータのDevの値
+     * @param array $devTabUrlInfo Devタブ内に表示するページのURL一覧
+     * @return string Devタブに表示するページのURL
+     * 
+     */
+    private function _getDevTabInfo($devParam, $devTabUrlInfo) {
+
+        // パラメータからDevタブのlistのキー値を作成する
+        $listkey = strtoupper($devParam);
+
+        // homeパラメータ値がDevタブに存在するurlと一致しない場合
+        if (!array_key_exists($listkey, $devTabUrlInfo)) {
+            // Dev_topを表示させる
+            return $devTabUrlInfo['TOP'];
+        } else {
+            return $devTabUrlInfo[$listkey];
+        }
+    }
+
+    /**
+     * Docsタブの情報を取得する
+     * 
+     * @param string $docsParam リクエストパラメータのdocsの値
+     * @param array $docsTabUrlInfo Docsタブ内に表示するページのURL一覧
+     * @return string Docsタブに表示するページのURL
+     * 
+     */
+    private function _getDocsTabInfo($docsParam, $docsTabUrlInfo) {
+
+        // パラメータからDocsタブのlistのキー値を作成する
+        $listkey = strtoupper($docsParam);
+
+        // homeパラメータ値がdocsタブに存在するurlと一致しない場合
+        if (!array_key_exists($listkey, $docsTabUrlInfo)) {
+            // docs_topを表示させる
+            return $docsTabUrlInfo['TOP'];
+        } else {
+            return $docsTabUrlInfo[$listkey];
+        }
+    }
+
+}
```

Comment:
__[code_sniffer]__
* The closing brace for the class must go on the next line after the body